core/src/toga/window.py

```diff
@@ -222,23 +236,37 @@ def position(self, position: tuple[int, int]) -> None:
         self._impl.set_position(position)
 
     def show(self) -> None:
-        """Show window, if hidden."""
+        """Show the window, if hidden."""
+
         if self.app is None:
-            raise AttributeError(
-                "Can't show a window that doesn't have an associated app"
-            )
+            # Needs to be a late import to avoid circular dependencies.
+            from toga import App
+
+            App.app.windows += self
+
```

Comment:
This would be a significant API change, but what do you think of explicitly passing the app in the constructor of the Window? The association between window and app would be more explicit and less magical to the user.

Reply:
I agree it would be more explicit - but given that App is a singleton, I'm not sure we gain much from that explicitness. It would also be a *huge* backwards incompatibility, compared to this change which is effectively invisible to existing usage.

Reply:
Ok, SGTM. I was just curious about the reasoning.